samples/matter/common/src/bridge/ble_connectivity_manager.cpp

```diff
@@ -79,10 +79,15 @@ void BLEConnectivityManager::ConnectionHandler(bt_conn *conn, uint8_t conn_err)
 		return;
 	}
 
-	LOG_INF("Connected: %s", str_addr);
-
 	/* TODO: Add security validation. */
 
+	if (conn_err && Instance().mRecovery.mRecoveryInProgress) {
+		Instance().mRecovery.PutDevice(device);
```

Comment:
I think it isn't because when I call the NotifyLostDevice method then I call PutDevice and start a timer, after the timer counter elapses I do scanning and decide whether the device was found or not, and from this point, I assign that recovery has been started. So I cannot move it there directly. We need to check it here because we shouldn't put the device if it is the first connection.